jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -179,6 +179,15 @@ def _on_glue_y_display_unit_changed(self, y_unit_str):
                 self.flux_unit.selected,
                 self.angle_unit.selected
             )
+            self.hub.broadcast(GlobalDisplayUnitChanged("sb", self.sb_unit_selected, sender=self))
+
+        else:
+            # if cube was loaded in flux units, we still need to broadcast
+            # a 'sb' message for mouseover info. this should be removed when
+            # unit change messaging is improved and is a temporary fix
+            self.hub.broadcast(GlobalDisplayUnitChanged('sb',
+                                                        self.flux_unit.selected,
```

Comment:
It worries me that this is saying the axis is `sb` but sending the selected flux unit...I guess it works but it seems misleading.

Reply:
yeah, the problem is that coords_info is ONLY listening to 'sb' messages, but it should also pick up when 'flux' is changed in case the loaded cube wasn't in surface brightness

Reply:
why is this the case?